src/wazuh_modules/syscollector/sysInfoWin.cpp

```diff
@@ -168,4 +189,137 @@ void SysInfo::getMemory(nlohmann::json& info) const
+            {
+                packageJson["version"] = value;
+            }
+            if (packageReg.string("Publisher", value))
+            {
+                packageJson["vendor"] = value;
+            }
+            if (packageReg.string("InstallDate", value))
+            {
+                packageJson["install_time"] = value;
+            }
+            if (packageReg.string("InstallLocation", value))
+            {
+                packageJson["location"] = value;
+            }
+            if (!packageJson.empty())
+            {
+                if (access & KEY_WOW64_32KEY)
+                {
+                    packageJson["architecture"] = "i686";
+                }
+                else if (access & KEY_WOW64_64KEY)
+                {
+                    packageJson["architecture"] = "x86_64";
+                }
+                else
+                {
+                    packageJson["architecture"] = "unknown";
+                }
+                data.push_back(packageJson);
+            }
+        }
+    }
+    catch(...)
+    {
+    }
+}
+
+static void getHotFixFromReg(const HKEY key, const std::string& subKey, nlohmann::json& data)
+{
+    try
+    {
+        std::set<std::string> hotfixes;
+        Utils::Registry root{key, subKey, KEY_WOW64_64KEY | KEY_ENUMERATE_SUB_KEYS | KEY_READ};
+        const auto packages{root.enumerate()};
+        for (const auto& package : packages)
+        {
+            if (Utils::startsWith(package, "Package_"))
+            {
+                std::string value;
+                Utils::Registry packageReg{key, subKey + "\\" + package, KEY_WOW64_64KEY | KEY_READ};
+                if (packageReg.string("InstallLocation", value))
+                {
+                    value = Utils::toUpperCase(value);
+                    const auto start{value.find("KB")};
+                    if (start != std::string::npos)
+                    {
+                        value = value.substr(start);
+                        const auto end{value.find("-")};
+                        value = value.substr(0, end);
+                        hotfixes.insert(value);
+                    }
+                }
+            }
+        }
+        for (const auto& hotfix : hotfixes)
+        {
+            data.push_back({{"hotfix", hotfix}});
+        }
+    }
+    catch(...)
+    {
+    }
+}
+
+static void getHotFixFromRegNT(const HKEY key, const std::string& subKey, nlohmann::json& data)
+{
+    static const std::string KB_PREFIX{"KB"};
+    try
+    {
+        std::set<std::string> hotfixes;
+        Utils::Registry root{key, subKey, KEY_WOW64_64KEY | KEY_ENUMERATE_SUB_KEYS | KEY_READ};
+        const auto packages{root.enumerate()};
+        for (const auto& package : packages)
+        {
+            auto value{Utils::toUpperCase(package)};
+            if (Utils::startsWith(value, KB_PREFIX))
+            {
+                value = value.substr(KB_PREFIX.size());
```

Comment:
Probably you can define a static const for KB_PREFIX.size() and use it here.

Reply:
will do